ow_lander/src/ow_actions/trajectory_planner.py

```diff
@@ -0,0 +1,999 @@
+# The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+# Research and Simulation can be found in README.md in the root directory of
+# this repository.
+
+import rospy
+import math
+import copy
+import moveit_commander
+from moveit_msgs.msg import PositionConstraint, RobotTrajectory
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+from tf.transformations import quaternion_from_euler, euler_from_quaternion
+from geometry_msgs.msg import Quaternion
+from shape_msgs.msg import SolidPrimitive
+from std_msgs.msg import Header
+from moveit_msgs.srv import GetPositionFK
+
+from ow_actions import constants
+from ow_actions.common import Singleton, is_shou_yaw_goal_in_range
+
+class ArmTrajectoryPlanner(metaclass = Singleton):
+
+  def __init__(self):
+    # basic moveit interface for arm control
+    self._robot = moveit_commander.RobotCommander()
+    self._move_arm = moveit_commander.MoveGroupCommander('arm')
+    self._move_grinder = moveit_commander.MoveGroupCommander('grinder')
+    # NOTE: never used
+    # self._move_limbs = moveit_commander.MoveGroupCommander('limbs')
+
+    # enable forward kinematic calculations
+    SERVICE_TIMEOUT = 30 # seconds
+    rospy.wait_for_service('compute_fk', SERVICE_TIMEOUT)
+    self._compute_fk_srv = rospy.ServiceProxy('compute_fk', GetPositionFK)
+
+  # def __del__(self):
+  #   # persistent service proxies must be manually closed
+  #   self._compute_fk_srv.close()
+
+  def compute_forward_kinematics(self, fk_target_link, robot_state):
+    # TODO: may raise ROSSerializationException
+    goal_pose_stamped = self._compute_fk_srv(
+      Header(0, rospy.Time.now(), "base_link")
+      [fk_target_link],
+      robot_state
+    )
+    return goal_pose_stamped.pose_stamped[0].pose
+
+  def plan_arm_to_target(self, target_name):
+    target_joints = self._move_arm.get_named_target_values(target_name)
+    self._move_arm.set_joint_value_target(target_joints)
+    # NOTE: _success, _planning_time, and _error are not used but are named in
+    #       case they are needed
+    _success, plan, _planning_time, _error = self._move_arm.plan()
+    return plan
+
+  ## DEPRECATED and logic disabled
```

Comment:
In the interest of completing this refactoring phase, suggest dealing with this later.  Arm stopping seems a bit complicated and worthy of separate attention.

Reply:
I've tested Stow and Unstow and find that:
 - The Stop action no longer works
 - Arm faults behave as though "arm_continues_in_fault" is selected.  I.e. the unfaulted joints keep moving.

Again I'd suggest a rework of this functionality be separate from refactoring, and even pushed into Release 12 unless something is broken.

Reply:
Just reiterating our discussion from this morning: Yes, things are broken in this branch, namely: stop arm action and stop arm actions on fault. I have an implementation ready for review once this review is approved for a merge. Followed by that will be an implementation of all of our ow_lander actions into this new framework.